### PR TITLE
Use dub from the release binary

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: posix.mak [hunk 1]
```diff
@@ -65,14 +63,14 @@ PHOBOS_STABLE_FILES_GENERATED := $(subst $(PHOBOS_STABLE_DIR), $(PHOBOS_STABLE_D
...
 STABLE_DMD_ROOT=$(GENERATED)/stable_dmd-$(STABLE_DMD_VER)
 STABLE_DMD_URL=http://downloads.dlang.org/releases/2.x/$(STABLE_DMD_VER)/dmd.$(STABLE_DMD_VER).$(OS).zip
-STABLE_DMD=$(STABLE_DMD_ROOT)/dmd2/$(OS)/$(if $(filter $(OS),osx),bin,bin$(MODEL))/dmd
+STABLE_DMD_BIN_ROOT=$(STABLE_DMD_ROOT)/dmd2/$(OS)/$(if $(filter $(OS),osx),bin,bin$(MODEL))
+STABLE_DMD=$(STABLE_DMD_BIN_ROOT)/dmd
 STABLE_DMD_CONF=$(STABLE_DMD).conf
-STABLE_RDMD=$(STABLE_DMD_ROOT)/dmd2/$(OS)/$(if $(filter $(OS),osx),bin,bin$(MODEL))/rdmd \
-	--compiler=$(STABLE_DMD) -conf=$(STABLE_DMD_CONF)
+STABLE_RDMD=$(STABLE_DMD_BIN_ROOT)/rdmd --compiler=$(STABLE_DMD) -conf=$(STABLE_DMD_CONF)
+DUB=$(STABLE_DMD_BIN_ROOT)/dub
```
Comment (wilzbach) [inline posix.mak]: `DUB` was the existing name, but for consistency it could be renamed to `DUB_STABLE`

File: posix.mak [hunk 2]
```diff
@@ -578,13 +573,19 @@ dpl-docs: ${DUB} ${STABLE_DMD}
...
-${STABLE_DMD}:
-	mkdir -p ${STABLE_DMD_ROOT}
-	TMPFILE=$$(mktemp deleteme.XXXXXXXX) && curl -fsSL ${STABLE_DMD_URL} > ${TMP}/$${TMPFILE}.zip && \
-		unzip -qd ${STABLE_DMD_ROOT} ${TMP}/$${TMPFILE}.zip && rm ${TMP}/$${TMPFILE}.zip
+# .tar.xz is archives are smaller (and don't need a temporary dir) -> prefer if available
+HAS_XZ := $(shell command -v xz 2> /dev/null)
+${STABLE_DMD_ROOT}/.downloaded:
+	@mkdir -p $(dir $@)
+ifdef HAS_XZ
+	curl -fSL --retry 3 $(subst .zip,.tar.xz,$(STABLE_DMD_URL)) | tar -Jxf - -C $(dir $@)
```
Comment (wilzbach) [inline posix.mak]: The `tar.xz` balls has about half the size of the zip archive and apart from making the CI generate less traffic on the AWS bucket, it's quite noticeably faster on slower connections.
If there's consensus about this, I would propose to do the same for the DMD bootstrapping.

File: posix.mak [hunk 3]
```diff
@@ -573,11 +573,17 @@ dpl-docs: ${DUB} ${STABLE_DMD}
 	DFLAGS="$(DPL_DOCS_DFLAGS)" ${DUB} build --root=${DPL_DOCS_PATH} \
 		--compiler=${STABLE_DMD}
 
+# .tar.xz is archives are smaller (and don't need a temporary dir) -> prefer if available
```
Comment (CyberShadow) [inline posix.mak]: I think there's an extra verb here

File: posix.mak (hunk 2 above)
Comment (CyberShadow) [inline posix.mak]: I wonder, how likely is it that `xz` won't be available on users' machines (and it would be too onerous to install it)?

**Comment by CyberShadow** [comment]
Not really a complaint but not sure why these two not-too-related changes are together in one PR :)

File: posix.mak (hunk 2 above)
Comment (wilzbach) [inline posix.mak]: I thought about the same, but then I guessed that if I didn't include the fallback, chances of complaint / rejection would be high.

I actually think that `zip` is less frequent than xz-utils

**Comment by wilzbach** [comment]
> Not really a complaint but not sure why these two not-too-related changes are together in one PR :)

Because they both touch this download command and my motivation/time was pretty low yesterday. I don't mind splitting them up if it's a controversial change - I didn't think that it would be.

**Comment by wilzbach** [comment]
> Because they both touch this download command and my motivation/time was pretty low yesterday. I don't mind splitting them up if it's a controversial change - I didn't think that it would be.

Alrighty - I guess I have to split into two PRs and rebase one then :/
-> https://github.com/dlang/dlang.org/pull/1850

**Comment by CyberShadow** [comment]
> Alrighty - I guess I have to split into two PRs and rebase one then :/

Sorry, I didn't mean to imply that you should do that.

File: posix.mak (hunk 1 above)
Comment (CyberShadow) [inline posix.mak]: (I'm guessing you meant `STABLE_DUB`)

**Pull request merged**